helm/prometheus-rules/templates/alerting-rules/app.rules.yml

```diff
@@ -143,7 +143,7 @@ spec:
     ## By default they go to Honeybadger, unless one of the teams below is configured.
     - alert: WorkloadClusterAppFailed
       annotations:
-        description: '{{`Workload Cluster App {{ $labels.name }}, version {{ $labels.version }} is {{if $labels.status }} in {{ $labels.status }} state. {{else}} not installed. {{end}}`}}'
+        description: '{{`Workload Cluster App {{ $labels.exported_namespace }}/{{ $labels.name }}, version {{ $labels.version }} is {{if $labels.status }} in {{ $labels.status }} state. {{else}} not installed. {{end}}`}}'
```

Comment:
I guess it's just `namespace` rather than `exported_namespace`.
At least from the labels I've had on `gauss` with this request.

Reply:
`$labels.namespace` is often overridden. This was the case for me in many alerts. Let's take [this alert](https://giantswarm.app.opsgenie.com/alert/detail/18c3ee0c-2a08-4603-be2f-701bd26a8bba-1661868811823/details) as an example. The `namespace` is `giantswarm`, but `exported_namespace` is `dmec3`, which is the cluster's name and the namespace in which the failing app can be found. The same applies to CAPX clusters, where `exported_namespace` equals organization namespace. That's the proper namespace where the App CR is.

Reply:
We decided in atlas to remove the exported_namespace so we need to fix the source of the metric

Reply:
So the issue is the app exporter already has a namespace label but a new one is added via relabelling which sets the existing namespace label to exported_namespace. We need to fix this

Reply:
I'll try this https://github.com/giantswarm/kyverno-policies-observability/pull/27

Reply:
Consider this PR on hold, then.

Reply:
PR works but kyverno is configured to ignore it's own namespace which is the giantswarm namespace and is excluding a lot of service monitors. As a suggestion until this is resolved, can we change the labels to:

```{{ if $labels.exported_namespace }}{{ $labels.exported_namespace }}{{ else }}{{ $labels.namespace }}{{ end }}/{{ $labels.name }}```

Reply:
Good suggestion, thanks! I'll update my PR.